core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,94 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final int lastEpisodeCount;
+    private final boolean shouldShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
+
+        FeedPreferences prefs = feed.getPreferences();
+        if (!prefs.getKeepUpdated() || !prefs.getShowEpisodeNotification()) {
+            shouldShowNotification = false;
+            lastEpisodeCount = -1;
+            return;
+        }
+
+        lastEpisodeCount = getNewEpisodeCount(feedId);
+        shouldShowNotification = true;
+    }
+
+    public void showIfNeeded(Context context, Feed feed) {
+        if (!shouldShowNotification) {
+            return;
+        }
+
+        long feedId = feed.getId();
+        int newEpisodes = getNewEpisodeCount(feedId);
+
+        if (newEpisodes > lastEpisodeCount) {
+            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+            showNotification(newEpisodes, feed, context, notificationManager);
+        }
+    }
+
+    private static void showNotification(int newEpisodes, Feed feed, Context context,
+                                         NotificationManagerCompat notificationManager) {
+        Resources res = context.getResources();
+        String text = res.getQuantityString(
+                R.plurals.new_episode_notification_message, newEpisodes, newEpisodes, feed.getTitle()
+        );
+        String title = res.getQuantityString(R.plurals.new_episode_notification_title, newEpisodes);
```

Comment:
> I personally dislike that solution, because i feel like I will be more disinformed as suposed to informed by that. But I guess I will implement it soon (DONE).

Maybe we could ask this question in the original issue. Also, the manual/automatic thing could be interesting to ask there.